src/app/app.datatypes.ts

```diff
@@ -63,14 +64,32 @@ export class GetAddressResponseTransaction {
   txid: string;
 }
 
-export function parseGetAddressTransaction(raw: GetAddressResponseTransaction): Transaction {
+export function parseGetAddressTransaction(raw: GetAddressResponseTransaction, address: string): Transaction {
+
+  //Detect if the address sent or received the coins.
+  let incoming: boolean = true;
+  if (raw.inputs[0].owner.toLowerCase() == address.toLowerCase()) { incoming = false; }
```

Comment:
You have to check for each input, as the input could also be a second or later element in the array